### PR TITLE
HPCC-31426 CentOS 7 + Python 3.8

### PR DESCRIPTION
**Pull request by GordonSmith**
Add custom build to support CentOS 7 + Python 3.8

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dockerfiles/vcpkg/build.sh [hunk 1]
```diff
@@ -46,30 +49,43 @@ function doBuild() {
...
-    if [ "$1" == "centos-7" ]; then
+    if [ "$1" == "centos-7*" ]; then
         CMAKE_OPTIONS_EXTRA="-DVCPKG_TARGET_TRIPLET=x64-centos-7-dynamic"
     elif [ "$1" == "amazonlinux" ]; then
         CMAKE_OPTIONS_EXTRA="-DVCPKG_TARGET_TRIPLET=x64-amazonlinux-dynamic"
     fi
     mkdir -p $HOME/.ccache
     docker run --rm \
         --mount source="$(pwd)",target=/hpcc-dev/HPCC-Platform,type=bind,consistency=cached \
+        --mount source="$(realpath ~)/.cache/vcpkg",target=/root/.cache/vcpkg,type=bind,consistency=cached \
```
Comment (ghalliday) [inline dockerfiles/vcpkg/build.sh:60]: What is this change for.  Worth documenting related changes in the jira.

File: dockerfiles/vcpkg/build.sh [hunk 2]
```diff
@@ -26,12 +27,14 @@ echo "DOCKER_PASSWORD: $DOCKER_PASSWORD"
...
+CMAKE_OPENBLAS_OPTIONS="-G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DHPCC_SOURCE_DIR=/hpcc-dev/HPCC-Platform -DCONTAINERIZED=OFF -DCPACK_STRIP_FILES=OFF -DECLBLAS=ON -DVCPKG_FILES_DIR=/hpcc-dev -DCPACK_THREADS=0 -DUSE_OPTIONAL=OFF -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
+CMAKE_PLATFORM_OPTIONS="-G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DHPCC_SOURCE_DIR=/hpcc-dev/HPCC-Platform -DCONTAINERIZED=OFF -DCPACK_STRIP_FILES=ON -DPLATFORM=ON -DVCPKG_FILES_DIR=/hpcc-dev -DCPACK_THREADS=0 -DUSE_OPTIONAL=OFF -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
 
 function doBuild() {
-    docker pull "hpccsystems/platform-build-base-$1:$VCPKG_REF" || true
-    docker pull "hpccsystems/platform-build-$1:$VCPKG_REF" || true
-    docker pull "hpccsystems/platform-build-$1:$GITHUB_BRANCH" || true
+    # docker pull "hpccsystems/platform-build-base-$1:$VCPKG_REF" || true
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply dockerfiles/vcpkg/build.sh:35]: FYI That build.sh file is only for local testing and used as part of any process - the pull should really be behind an arg flag, I will fix.

File: dockerfiles/vcpkg/build.sh (hunk 1 above)
Comment (GordonSmith) [reply dockerfiles/vcpkg/build.sh:60]: Moves the cached vcpkg build location into the host machine, which in turn will speed up local calls to build.sh for any "vcpkg_overlay" contents.

File: dockerfiles/vcpkg/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (GordonSmith) [reply dockerfiles/vcpkg/build.sh]: I rolled back this change

File: dockerfiles/vcpkg/build.sh (hunk 1 above)
Comment (GordonSmith) [reply dockerfiles/vcpkg/build.sh:60]: I added the change information into the commit message.

**Review by ghalliday — APPROVED** [review]
@Michael-Gardner please take a look

**Review by Michael-Gardner — COMMENTED** [review]
Looks good. Code looks fine. I was having issues updating the vcpkg submodule for this branch when I was trying to test it, but I think that's due to what I have cached locally.

**Pull request merged**